### DISCUSSION TITLE
How do I automatically mine blocks when I fork the mainnet?

### DISCUSSION THREAD

**Discussion by the-content** [NomicFoundation/hardhat#4378] [General]
I am using hardhat's fork feature locally, I forked BSC's mainnet, and in my contract, I need to display the calculated gain based on block.timestamp in real time in the dapp based on the current block timestamp, but I noticed that the timestamp doesn't automatically increment if there is no transaction occurring, is there any way to do this? Or is there any config item to make the fork function tap automatically?

**Comment by fvictorio** [comment]
You could enable [interval mining](https://hardhat.org/hardhat-network/docs/reference#mining-modes), or manually mine new blocks with the [mine helper](https://hardhat.org/hardhat-network-helpers/docs/reference#mine([blocks],-[options])).

**Reply by the-content** [reply]
Thank you very much, I will try it
